packages/combo-box/src/vaadin-combo-box-data-provider-mixin.js

```diff
@@ -85,6 +85,10 @@ export const ComboBoxDataProviderMixin = (superClass) =>
     ready() {
       super.ready();
       this._scroller.addEventListener('index-requested', (e) => {
+        if (!this._shouldFetchData()) {
```

Comment:
The tests pass without this check, but I decided to add anyway for the sake of safety.